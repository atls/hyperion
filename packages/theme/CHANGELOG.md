# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.0.10 (2021-03-08)

**Note:** Version bump only for package @atlantis-lab/theme





## 0.0.9 (2021-02-28)

**Note:** Version bump only for package @atlantis-lab/theme





## 0.0.8 (2021-02-26)

**Note:** Version bump only for package @atlantis-lab/theme





## 0.0.7 (2021-02-26)

**Note:** Version bump only for package @atlantis-lab/theme





## 0.0.6 (2021-02-26)

**Note:** Version bump only for package @atlantis-lab/theme





## 0.0.5 (2021-02-26)

**Note:** Version bump only for package @atlantis-lab/theme





## 0.0.4 (2021-02-26)

**Note:** Version bump only for package @atlantis-lab/theme





## 0.0.3 (2021-02-26)

**Note:** Version bump only for package @atlantis-lab/theme





## 0.0.2 (2021-02-22)

**Note:** Version bump only for package @atlantis-lab/theme





## 0.0.1 (2021-02-15)

**Note:** Version bump only for package @atlantis-lab/theme
