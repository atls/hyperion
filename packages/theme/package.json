{
  "name": "@atlantis-lab/theme",
  "repository": "git@github.com:Atlantis-Lab/uikit.git",
  "version": "0.0.25",
  "license": "MIT",
  "main": "dist/index.js",
  "source": "src/index.ts",
  "typings": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "clean": "rm -r -f dist",
    "prepublishOnly": "yarn run build",
    "build": "tsc -p tsconfig.build.json",
    "prepack": "pubflow store",
    "postpack": "pubflow restore"
  },
  "dependencies": {
    "@emotion/css": "^11.1.3",
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.1.5",
    "react": "17.0.1",
    "styled-system": "^5.1.5"
  },
  "devDependencies": {
    "@atlantis-lab/design": "^0.0.2",
    "@atlantis-lab/theme": "0.0.22",
    "@emotion/react": "^11.1.5",
    "@storybook/react": "6.1.21",
    "@types/styled-system": "5.1.10",
    "react-test-renderer": "^17.0.1"
  },
  "publishConfig": {
    "access": "public"
  }
}
