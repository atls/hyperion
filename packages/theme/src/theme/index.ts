export * from './fontWeights'
export * from './lineHeights'
export * from './shadows'
export * from './fontFaces'
export * from './fontSizes'
export * from './borders'
export * from './colors'
export * from './fonts'
export * from './radii'
