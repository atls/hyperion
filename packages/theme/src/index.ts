import * as theme from './theme'

export * from './Provider'
export * from './globals'

export { theme }
