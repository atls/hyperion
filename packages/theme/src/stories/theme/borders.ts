import { Borders } from '../../theme'

export const borders: Borders = {}
