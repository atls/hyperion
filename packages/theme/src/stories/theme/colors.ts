import { Colors } from '../../theme'

export const colors: Colors = {}
