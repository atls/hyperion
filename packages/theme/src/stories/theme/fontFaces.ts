import { FontFaces } from '../../theme'

export const fontFaces: FontFaces = []
