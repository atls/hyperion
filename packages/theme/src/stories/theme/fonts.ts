import { Fonts } from '../../theme'

export const fonts: Fonts = {}
