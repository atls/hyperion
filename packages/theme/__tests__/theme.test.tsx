import React                                 from 'react'
import renderer                              from 'react-test-renderer'

import { useTheme }                          from '@emotion/react'

import * as actualTheme                      from './theme/theme'
import { ThemeProvider, injectGlobalStyles } from './theme'

describe('test suit for theme', () => {
  it('should render ThemeProvider correctly', () => {
    const ThemeConsumer = () => {
      const theme = useTheme()

      expect(theme).toMatchObject(actualTheme)

      return <p>Inside provider</p>
    }

    const tree = renderer
      .create(
        <ThemeProvider>
          <ThemeConsumer />
        </ThemeProvider>,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should injectGlobalStyles with no errors', () => {
    expect(injectGlobalStyles()).toBe(undefined)
  })
})
