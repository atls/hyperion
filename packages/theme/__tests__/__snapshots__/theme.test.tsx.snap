// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suit for theme should render ThemeProvider correctly 1`] = `
<p>
  Inside provider
</p>
`;
